store/splitstore.go

```diff
@@ -0,0 +1,93 @@
+package store
+
+import (
+	"github.com/loomnetwork/go-loom/plugin"
+)
+
+type splitStore struct {
+	KVReader
+	VersionedKVStore
+	deleted map[string]bool
+	version int64
+}
+
+func NewSplitStore(full KVReader, empty VersionedKVStore, version int64) VersionedKVStore {
+	return &splitStore{
+		KVReader:         full,
+		VersionedKVStore: empty,
+		deleted:          make(map[string]bool),
+		version:          version,
+	}
+}
+
+func (ss *splitStore) Get(key []byte) []byte {
+	if ss.VersionedKVStore.Has(key) {
+		return ss.VersionedKVStore.Get(key)
+	}
+	if ss.deleted[string(key)] {
+		return nil
+	}
+	return ss.KVReader.Get(key)
+}
+
+func (ss *splitStore) Range(prefix []byte) plugin.RangeData {
+	readerRange := ss.KVReader.Range(prefix)
+	updateRange := ss.VersionedKVStore.Range(prefix)
+	for _, re := range updateRange {
+		if !ss.KVReader.Has(re.Key) && !ss.deleted[string(re.Key)] {
+			readerRange = append(readerRange, re)
```

Comment:
This `range` function does not seem right to me. The logic is appending the data instead of updating it. However, I believe it is not used in evm logic but probably we should try to get it right.